File: 160-intersection-of-two-linked-lists/intersection-of-two-linked-lists.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def getIntersectionNode(self, headA: ListNode, headB: ListNode) -> Optional[ListNode]:
        p1=headA
        p2=headB
        cnt=0
        while True:
            if p1==p2:
                return p1

            p1=p1.next
            p2=p2.next

            if(p2==None):
                cnt+=1
                p2=headA
            if(p1==None):
                p1=headB    
            if cnt > 1:
                return None

File: 160-intersection-of-two-linked-lists/intersection-of-two-linked-lists.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode *getIntersectionNode(ListNode *headA, ListNode *headB) {



        ListNode *p1=headA;
        ListNode *p2=headB;
        int cnt=0;
        while(true)
        {

            if(p1==p2)
            return p1;
            p1=p1->next;
            p2=p2->next;
            if(p1==nullptr){
            p1=headB;
            cnt++;
            }
            if(p2==nullptr)
            p2=headA;
            if(cnt>1)
            return nullptr;
        }


        
    }
};